### PR TITLE
fix(docs): improve WordPress tutorial

### LINKED ISSUE #20080: Wordpress tutorial is broken?
## Description

Following the wordpress tutorial results in errors.

### Steps to reproduce

follow this tutorial exactly from a fresh start:
https://www.gatsbyjs.org/tutorial/wordpress-source-plugin-tutorial/

### Expected result

the tutorial should use the gatsby-source-wordpress to source content from the default wordpress site.

### Actual result

here is the results from a `gatsby build` after adding the tutorials example `gastsby-config.js`:

https://gist.github.com/joshuacox/22a34a8b88c64842438ebec1a09a604d

### Environment

```

  System:
    OS: Linux 5.4 Arch Linux
    CPU: (8) x64 Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
    Shell: 5.7.1 - /usr/bin/zsh
  Binaries:
    Node: 10.16.3 - ~/.nvm/versions/node/v10.16.3/bin/node
    Yarn: 1.19.2 - ~/.nvm/versions/node/v10.16.3/bin/yarn
    npm: 6.9.0 - ~/.nvm/versions/node/v10.16.3/bin/npm
  Languages:
    Python: 3.8.0 - /usr/bin/python
  Browsers:
    Firefox: 71.0
  npmPackages:
    gatsby: ^2.18.8 => 2.18.8 
    gatsby-image: ^2.2.34 => 2.2.34 
    gatsby-plugin-manifest: ^2.2.31 => 2.2.31 
    gatsby-plugin-offline: ^3.0.27 => 3.0.27 
    gatsby-plugin-react-helmet: ^3.1.16 => 3.1.16 
    gatsby-plugin-sharp: ^2.3.5 => 2.3.5 
    gatsby-source-filesystem: ^2.1.40 => 2.1.40 
    gatsby-source-wordpress: ^3.1.53 => 3.1.53 
    gatsby-transformer-sharp: ^2.3.7 => 2.3.7 
  npmGlobalPackages:
    gatsby-cli: 2.8.14
```

### PR DESCRIPTION
**Pull request by TylerBarnes**
Closes #20080 

I went through the tutorial locally and followed all the steps and made updates as I went through, then afterwards I followed all my updated steps again from scratch to make sure it's all working properly.

I noticed the demo url `dev-gatbsyjswp.pantheonsite.io` didn't actually work anymore. Since that was a dev site, pantheon froze it after a period of inactivity. I've launched the demo site to `live-gatbsyjswp.pantheonsite.io` and updated the URL here, that one wont freeze if it's inactive. I'm not sure how long the dev- site has been down for.

I also noticed the README for `gatsby-source-wordpress` had a WordPress.org url listed in the plugin options example, but lower down had `hostingWPCOM: false`. I updated that url to `live-gatbsyjswp.pantheonsite.io` for consistency as that site is a WordPress.org site.

### CODE REVIEW

File: docs/tutorial/wordpress-source-plugin-tutorial.md [hunk 1]
```diff
@@ -32,15 +32,17 @@ cd wordpress-tutorial-site
 Install the `gatsby-source-wordpress` plugin. For extra reading on the plugin’s features and examples of GraphQL queries not included in this tutorial, see the [`gatsby-source-wordpress` plugin’s README file](/packages/gatsby-source-wordpress/?=wordpress).
 
 ```shell
-npm install --save gatsby-source-wordpress
+npm install gatsby-source-wordpress
```
[the remark this answers is not in the document]
Comment (TylerBarnes) [reply docs/tutorial/wordpress-source-plugin-tutorial.md]: True! I agree it makes sense to remove it wherever it's written that way. It's just a tiny bit more friction for new people. Another thing to understand when each new thing might already feel like a lot.

File: docs/tutorial/wordpress-source-plugin-tutorial.md [hunk 2]
```diff
@@ -54,11 +56,11 @@ module.exports = {
       options: {
         /*
          * The base URL of the WordPress site without the trailingslash and the protocol. This is required.
-         * Example : 'dev-gatbsyjswp.pantheonsite.io' or 'www.example-site.com'
+         * Example : 'demo.wp-api.org' or 'www.example-site.com'
```
[the remark this answers is not in the document]
Comment (TylerBarnes) [reply docs/tutorial/wordpress-source-plugin-tutorial.md]: Yeah, I found it in the WP REST docs when the dev- pantheon site wasn't working. I first started updating the tutorial to use it, and then realized we needed the pantheon site as the data is too different. It's nice that it exists though!

**Comment by TylerBarnes** [comment]
Thanks for mentioning that README @gillkyle . I updated it and removed the "important note" at the end as it doesn't seem to apply anymore.

File: docs/tutorial/wordpress-source-plugin-tutorial.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (muescha) [reply docs/tutorial/wordpress-source-plugin-tutorial.md]: i open a Issue for this:
- #22444 `docs: Suggestion: unify npm / remove --save from npm install`

**Pull request merged**